### PR TITLE
Cataloguing tool

### PR DESCRIPTION
**Pull request by redlibrarian**
This PR builds the feature specified in #1179

### CODE REVIEW

File: app/views/catalog/_index_default.html.erb [hunk 1]
```diff
@@ -23,7 +23,7 @@
           <% if document["held_by_tesim"] %>
             <% @held_by_locations = [] %>
             <% document["held_by_tesim"].first.split.each do |held_by| %>
-              <% @ual = true if @library_codes[held_by.to_i].include? "University of Alberta" %>
+              <% @ual = true if (@library_codes[held_by.to_i].include?("University of Alberta") or (@library_codes[held_by.to_i].include?("NEOS Free"))) %>
```
Comment (pgwillia) [inline app/views/catalog/_index_default.html.erb]: Looks like the changes from #1152 are duplicated here.

File: Gemfile [hunk 2]
```diff
@@ -59,6 +59,7 @@ group :test, :development do
   gem 'rspec-solr'
   gem 'simplecov', :require => false
   gem 'factory_bot'
+  gem 'zoom'
```
Comment (pgwillia) [inline Gemfile]: https://rubygems.org/gems/zoom -- oh cool, a ruby binding for z39.50

The installation for the [dependencies](https://www.rubydoc.info/gems/zoom/0.5.0) (requires [yaz](https://www.indexdata.com/resources/software/yaz/) with `--enabled-shared` option?) might be unexpected and are required by the change to the Gemfile.  Maybe add some tips, instructions or references to the readme?

Comment (pgwillia) [reply Gemfile]: This is how avalon handles it https://github.com/ualbertalib/avalon/blob/master/Gemfile#L162-L166

Comment (pgwillia) [reply Gemfile]:
```
git clone https://github.com/indexdata/yaz.git
cd yaz/
sudo apt-get install autoconf automake libtool gcc bison tclsh xsltproc docbook docbook-xml docbook-xsl libxslt1-dev libreadline-dev libwrap0-dev pkg-config libicu-dev make
./buildconf.sh
./configure --enable-shared
make
make install
```
It seems like there are packages available for ubuntu, etc.  Tried a few but didn't seem to work. Went the git route and built from scratch. Then `bundle install` worked for me for zoom.

Comment (pgwillia) [reply Gemfile]: Missing logic to create `tmp/records` if it's missing.

Comment (pgwillia) [reply Gemfile]: The index portion seems to work. There logging has some errors in it but the document with id:tmp1002481 appears in the Solr index. Then the redirect has a problem.
```
Redirected to http://://localhost:3000/catalog/tmp1002481
Completed 302 Found in 2895ms (ActiveRecord: 0.1ms)
[2018-07-12 14:52:03] ERROR URI::InvalidURIError: the scheme http does not accept registry part: : (or bad hostname?)
	/home/pjenkins/.rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/uri/generic.rb:1203:in `rescue in merge'
	/home/pjenkins/.rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/uri/generic.rb:1200:in `merge'
	/home/pjenkins/.rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/webrick/httpresponse.rb:276:in `setup_header'
	/home/pjenkins/.rvm/gems/ruby-2.1.5@discovery/gems/rack-1.6.10/lib/rack/handler/webrick.rb:16:in `setup_header'
	/home/pjenkins/.rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/webrick/httpresponse.rb:206:in `send_response'
	/home/pjenkins/.rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/webrick/httpserver.rb:110:in `run'
	/home/pjenkins/.rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/webrick/server.rb:295:in `block in start_thread'
```
Trying to visit that location by hand results in
```
Started GET "/catalog/tmp1002481" for 127.0.0.1 at 2018-07-12 14:53:32 -0600
Processing by CatalogController#show as HTML
  Parameters: {"id"=>"tmp1002481"}
DEPRECATION WARNING: Configuration is now done via blacklight.yml. Suppport for solr.yml will be removed in blacklight 6.0.0. (called from connection_config at /home/pjenkins/.rvm/gems/ruby-2.1.5@discovery/gems/blacklight-5.15.0/lib/blacklight.rb:83)
DEPRECATION WARNING: Configuration is now done via blacklight.yml. Suppport for solr.yml will be removed in blacklight 6.0.0. (called from connection_config at /home/pjenkins/.rvm/gems/ruby-2.1.5@discovery/gems/blacklight-5.15.0/lib/blacklight.rb:84)
Solr query: get select {:qt=>"document", :id=>"tmp1002481"}
Solr fetch (157.6ms)
Completed 500 Internal Server Error in 176ms (ActiveRecord: 0.1ms)

SymphonyService::Error::InvalidIdError - SymphonyService::Error::InvalidIdError:
  app/services/symphony_service.rb:24:in `initialize'
  app/helpers/holdings_helper.rb:12:in `holdings'
  app/controllers/catalog_controller.rb:40:in `show'
```

File: .env_deployment [hunk 3]
```diff
@@ -0,0 +1,13 @@
+RAILS_ENV=uat
```
[the remark this answers is not in the document]
Comment (redlibrarian) [reply .env_deployment]: Hm. I've done this, but it doesn't seem to be recognizing that this change has been made... not sure why.

**Comment by redlibrarian** [comment]
Hm. The check is failing on building the extensions for 'zoom'. @pgwillia I can't see your comment about how avalon handles zoom anymore - did you mean using [this](https://github.com/ualbertalib/avalon/blob/master/Gemfile#L165) instead of the rubygems version?

**Comment by pgwillia** [comment]
Replying to https://github.com/ualbertalib/discovery/pull/1180#issuecomment-407866042 

I meant guarding the gem from attempting to be installed in environments where it's not required (like test)
```
group :development, :production, :staging do
    #   NOTE: Requires the yaz library to be installed
    gem 'zoom'
end

**Comment by redlibrarian** [comment]
@pgwillia OK, thanks. I'll add that; I'm not sure what to do about the extension-build problem in Travis.

**Comment by pgwillia** [comment]
Replying to https://github.com/ualbertalib/discovery/pull/1180#issuecomment-407872882
Travis-ci uses RAILS_ENV=test so the gem wouldn't be built and the dependencies not being installed wouldn't matter.

**Comment by pgwillia** [comment]
Replying to https://github.com/ualbertalib/discovery/pull/1180#issuecomment-407884252

In `.travis.yml` you'll have to make a small change to ignore staging as well
```
-bundler_args: --without development production --jobs=3 --retry=3
+bundler_args: --without development staging production --jobs=3 --retry=3
```

**Comment by redlibrarian** [comment]
@pgwillia Ah, OK - got it. Will that be a problem for anything else? I'd like to be able to try this out in staging...

**Comment by pgwillia** [comment]
Replying to https://github.com/ualbertalib/discovery/pull/1180#issuecomment-407896432

It might, I don't think @nmacgreg is done with moving test -> staging. Will probably also need to consult with him about installing the dependencies.

**Comment by redlibrarian** [comment]
@pgwillia I just discovered Neil is on vacation until after I leave, so I'm not sure now about deploying this change at all... I'll have to think about it.

File: app/controllers/cataloguing_controller.rb [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    record.fields.delete(MARC::ControlField.new('001',catkey))
+    record.fields.push MARC::ControlField.new('001', "tmp#{catkey}")
+    writer = MARC::Writer.new("#{Rails.root}/tmp/records/tmp#{catkey}.mrc")
+    writer.write(record)
+    writer.close
+    # delete old marc file
+
+  end
+
+  def index_record(catkey)
```
[the remark this answers is not in the document]
Comment (redlibrarian) [reply app/controllers/cataloguing_controller.rb]: OK, will do.

**Comment by pgwillia** [comment]
Moved this work to https://github.com/ualbertalib/discovery/tree/cataloguing_tool.  More information here https://github.com/ualbertalib/NEOSDiscovery/issues/218

**Pull request closed without merging**